### PR TITLE
New feature and bug fixing

### PR DESCRIPTION
**Pull request by hihelloluigi**
Hi, I made some changes on your project. _I hope they can be fine_

Syntax refactoring:
- I added 3 folders in pod project to sort in a better way the classes.
- I added some markers in your code to improve code legibility and division.

Functions added:
- Make optional customize cell delegate
- Create new delegate that return the date (I prefer this method instead of action button)
- Add max date. I read in some old issues that this feature was added but I didn't find it in last release.
- Make date formatter extensions public so user can customize date format

Bug corrected:
- When you selected time interval 1 and trying to select from 56 to 59 minutes the picker scroll to 55.
- When you selected a month with 31 or 30 days and switch to a month with less days, the picker show first days of month and return a wrong date for example (30/02/2018)

I know that are a lot of code changed so take all the time you need to check that everything works.

### CODE REVIEW

File: DayDatePicker/Classes/DayDatePicker/DayDatePickerView.swift [hunk 1]
```diff
@@ -277,45 +281,54 @@ extension DayDatePickerView : UITableViewDataSource, UITableViewDelegate {
...
         cell.textLabel?.textColor = textColor
-
+        
         if tableView == dayTableView {
             let date = Date(year: year, month: month, day: dayRange.lowerBound + indexPath.row)
             if let minDate = minDate, date < minDate {
                 cell.textLabel?.textColor = UIColor.lightGray
             }
-
+
```
Comment (hughbe) [inline DayDatePicker/Classes/DayDatePicker/DayDatePickerView.swift]: Can you revert these whitespace changes?

File: DayDatePicker/Classes/TimePicker/TimePickerView.swift [hunk 2]
```diff
@@ -353,40 +357,40 @@ extension TimePickerView {
...
-            setMinTime(hour: newValue, minute: _maxTime?.minute ?? 0, animated: true)
+            setMinTime(hour: newValue, minute: _maxTime?.minute ?? kMaxMinute, animated: true)
         }
     }
     
     @IBInspectable
     public var maxMinute: NSInteger {
         get {
-            return _maxTime?.minute ?? 0
+            return _maxTime?.minute ?? kMaxMinute
```
Comment (hughbe) [inline DayDatePicker/Classes/TimePicker/TimePickerView.swift]: can you get rid of the prefix "k" for all of these

**Comment by hughbe** [comment]
Some comments that I'd appreciate you fixing but this looks great! Thanks!

File: DayDatePicker/Classes/TimePicker/TimePickerView.swift (hunk 2 above)
Comment (hihelloluigi) [reply DayDatePicker/Classes/TimePicker/TimePickerView.swift]: What is your preferred syntax to write constant?

Comment (hughbe) [reply DayDatePicker/Classes/TimePicker/TimePickerView.swift]: Capitalized but without the prefix: e.g. `MaxMinute`

File: DayDatePicker/Classes/DayDatePicker/DayDatePickerView.swift (hunk 1 above)
Comment (hihelloluigi) [reply DayDatePicker/Classes/DayDatePicker/DayDatePickerView.swift]: I'm reading that probably is some of my git setting. Sorry 😓

**Pull request merged**